AndroidSDKCore/src/main/java/com/leanplum/Leanplum.java

```diff
@@ -718,27 +718,20 @@ public void response(List<Map<String, Object>> requests, JSONObject response, in
     LeanplumInternal.triggerStartIssued();
   }
 
-  private static void handleApiResponse(JSONObject response, List<Map<String, Object>> requests,
+  private static void handleStartResponse(JSONObject response, List<Map<String, Object>> requests,
                                         final RequestOld request, int countOfUnsentRequests) {
     JSONObject lastStartResponse = null;
 
-    // Find and handle the last start response.
     try {
-      // Checks if START event inside the current batch. If database index of START event bigger
-      // then a number of count of events that we got from the database - decrease START event
```

Comment:
Can you please explain what is happening here and how deleting this code helps?

Reply:
The issue can be reproduced when another request is made (e.g. `getVars`) while `start` didn't receive a response yet.
In such cases the request’s `dataBaseIndex` is 1 and `countOfUnsentRequests` is also 1:
![image001](https://user-images.githubusercontent.com/431167/71765393-d9447500-2efc-11ea-99b4-b193915d8c70.png)
This causes `return` to get called and the response of `start` is discarded. This will cause the callback for `start` never to get fired, causing the app to wait indefinitely for the callback to get fired.

The Zendesk ticket contains a more details description of the issue.